plugins/woocommerce/src/Internal/Admin/ReviewsListTable.php

```diff
@@ -0,0 +1,1393 @@
+					<?php echo esc_html( $post_type_object->labels->view_item ); ?>
+				</a>
+				<span class="post-com-count-wrapper post-com-count-<?php echo esc_attr( $product_post->ID ); ?>">
+					<?php $this->comments_bubble( $product_post->ID, get_pending_comments_num( $product_post->ID ) ); ?>
+				</span>
+			</div>
+			<?php
+
+		endif;
+
+		echo $this->filter_column_output( 'response', ob_get_clean(), $item ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	}
+
+	/**
+	 * Renders the type column.
+	 *
+	 * @param WP_Comment $item Review or reply being rendered.
+	 * @return void
+	 */
+	protected function column_type( $item ) {
+
+		$type = 'review' === $item->comment_type
+			? '&#9734;&nbsp;' . __( 'Review', 'woocommerce' )
+			: __( 'Reply', 'woocommerce' );
+
+		echo $this->filter_column_output( 'type', esc_html( $type ), $item ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	}
+
+	/**
+	 * Renders the rating column.
+	 *
+	 * @param WP_Comment $item Review or reply being rendered.
+	 * @return void
+	 */
+	protected function column_rating( $item ) {
+		$rating = get_comment_meta( $item->comment_ID, 'rating', true );
+
+		ob_start();
+
+		if ( ! empty( $rating ) && is_numeric( $rating ) ) {
+			$rating = (int) $rating;
+
+			$accessibility_label = sprintf(
+				/* translators: 1: number representing a rating */
+				__( '%1$s out of 5', 'woocommerce' ),
+				$rating
+			);
+
+			$stars = str_repeat( '&#9733;', $rating );
+			$stars .= str_repeat( '&#9734;', 5 - $rating );
+
+			?>
+			<span aria-label="<?php echo esc_attr( $accessibility_label ); ?>"><?php echo esc_html( $stars ); ?></span>
+			<?php
+		}
+
+		echo $this->filter_column_output( 'rating', ob_get_clean(), $item ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	}
+
+	/**
+	 * Renders any custom columns.
+	 *
+	 * @param WP_Comment $item        Review or reply being rendered.
+	 * @param string     $column_name Name of the column being rendered.
+	 * @return void
+	 */
+	protected function column_default( $item, $column_name ) {
+
+		ob_start();
+
+		/**
+		 * Fires when the default column output is displayed for a single row.
+		 *
+		 * This action can be used to render custom columns that have been added.
+		 *
+		 * @param WP_Comment $item The review or reply being rendered.
+		 */
+		do_action( 'woocommerce_product_reviews_table_column_' . $column_name, $item );
+
+		echo $this->filter_column_output( $column_name, ob_get_clean(), $item ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	}
+
+	/**
+	 * Runs a filter hook for a given column content.
+	 *
+	 * @param string     $column_name The column being output.
+	 * @param string     $output      The output content (may include HTML).
+	 * @param WP_Comment $item        The review or reply being rendered.
+	 * @return string
+	 */
+	protected function filter_column_output( $column_name, $output, $item ) {
```

Comment:
```suggestion
	protected function filter_column_output( string $column_name, string $output, WP_Comment $item ) {
```

I think we can add these.

Reply:
@agibson-godaddy any reasons for typed params in this specific method? From the `column_cb` method to the last one, params are not typed.

<img width="923" alt="image" src="https://user-images.githubusercontent.com/72993300/164714220-37679a97-cb7a-4a0b-b6f8-611fdd4f4ae3.png">

Reply:
I'm limiting to ones that are methods entirely build by us. So not doing it for:

1. Methods defined in the base class that we're overriding. (Because the original methods aren't typed.)
2. Methods that WordPress core is passing data to (in which case we don't directly control the value going in).

This one recommendation is a method we build ourselves and we pass data into solely ourselves.

Reply:
`review_type_dropdown()` could be typed as well, I guess.

Reply:
But if you see those methods that are not built by us, they call `filter_column_output` and are not typed, meaning that they can be null (not saying they will, but they can). I'm afraid that typing those parameters could result in exceptions due to how WP/WC call those methods. 😕 Not a strong opinion, though, just being safe.

Maybe other can chime in and help us decide (@godaddy-wordpress/mwc-engineering-experiences).

Reply:
@acastro1-godaddy 

There are indeed more methods where we could add strict param types or return types. 

Callbacks that handle arguments passed by hooks should have a `@param <type>|mixed` in the phpdoc and no strict types whatsoever. This is the approach we have in MWC code as well. 

If someone passes the wrong argument to a hook, then the responsibility is on the third party code.

Since the next release of WC will require 7.2 we can also support nullable types, I'd argue. 

I am not sure if A8C will enforce particular rules with regards to types... The codebase in /src doesn't seem consistent enough to tell.

I propose we apply strict types following the points brought up by @agibson-godaddy above and if A8C doesn't like something we can change that after we submit our PR for review by them.